File: ontrack-service/src/test/java/net/nemerosa/ontrack/service/support/property/TestExtensionFeature.java
package net.nemerosa.ontrack.service.support.property;

import net.nemerosa.ontrack.model.extension.ExtensionFeature;
import org.springframework.stereotype.Component;

@Component
public class TestExtensionFeature implements ExtensionFeature {
    @Override
    public String getId() {
        return "test";
    }

    @Override
    public String getName() {
        return "Test extension";
    }

    @Override
    public String getDescription() {
        return "Extensions for tests";
    }
}

File: ontrack-extension-api/src/testFixtures/kotlin/net/nemerosa/ontrack/extension/api/support/TestExtensionFeature.kt
package net.nemerosa.ontrack.extension.api.support

import net.nemerosa.ontrack.model.extension.ExtensionFeature
import org.springframework.stereotype.Component

@Component
class TestExtensionFeature : ExtensionFeature {
    override val id: String = "test"

    override val name: String = "Test extension"

    override val description: String = "Extensions for tests"

    override val version: String = "test"
}